go/createmetastoreentry_test.go

```diff
@@ -0,0 +1,39 @@
+package bucketclient_test
+
+import (
+	"github.com/jarcoal/httpmock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"io"
+	"net/http"
+
+	"github.com/scality/bucketclient/go"
+)
+
+var _ = Describe("CreateMetastoreEntry()", func() {
+	It("creates/updates a new metastore entry", func(ctx SpecContext) {
+		httpmock.RegisterResponder(
+			"POST", "/default/metastore/db/my-bucket",
+			func(req *http.Request) (*http.Response, error) {
+				defer req.Body.Close()
+				Expect(io.ReadAll(req.Body)).To(Equal(
+					[]byte(`{"name":"bucket-03","attributes":"{\"name\":\"bucket-03\"}",` +
+						`"creating":false,"deleting":false,"id":"bucket-03",` +
+						`"raftSessionID":1,"version":2,"raftSession":"rs-1"}`)))
+				return httpmock.NewStringResponse(200, ""), nil
+			},
+		)
+		Expect(client.CreateMetastoreEntry(ctx,
+			"my-bucket", bucketclient.MetastoreEntry{
+				Name:          "bucket-03",
+				Attributes:    `{"name":"bucket-03"}`,
+				Creating:      false,
+				Deleting:      false,
+				ID:            "bucket-03",
+				RaftSessionID: 1,
+				Version:       2,
+				RaftSession:   "rs-1",
```

Comment:
Instead of manually hardcoding the expected struct in the Expect call, we should be able to directly reuse the `testGetMetastoreEntryResponse` string by unmarshaling it into a `bucketclient.MetastoreEntry`. This will reduce duplication and make the test easier to maintain.

Something like this

```
var expectedMetastoreEntry bucketclient.MetastoreEntry
err := json.Unmarshal([]byte(testGetMetastoreEntryResponse), &expectedMetastoreEntry)
Expect(err).To(BeNil())

Expect(client.GetMetastoreEntry(ctx, "my-bucket")).To(Equal(expectedMetastoreEntry))
```

Reply:
It could work indeed, but from a testing perspective I like to be able to check that what I get is what I expect to get in details, esp. with such a small test case. Unmarshaling in the test case would prevent to see issues with the unmarshaling itself (even though Go's type safety make it less likely to happen, it's still possible, e.g. if the `json:...` attributes in the `MetastoreEntry` struct are incorrect).